### DISCUSSION TITLE
Dealing with class imbalance

### DISCUSSION THREAD

**Discussion by DarylWM** [MIC-DKFZ/nnUNet#1753] [Q&A]
Hello.

My dataset is class-imbalanced; one of my foreground classes is severely under-represented compared to the other two. Here is a typical class histogram:

<img width="280" alt="image" src="https://github.com/MIC-DKFZ/nnUNet/assets/1016303/c265399d-a091-4200-abba-b7e1a5695f71">

After training several configurations and ensembling, the confusion matrix of the model's predictions reflects the imbalance. The model performs well on the better-represented classes.

![image](https://github.com/MIC-DKFZ/nnUNet/assets/1016303/b0b5f990-ca75-4fbd-8952-dc3c5565360c)

I've looked at the trainer variant nnUNetTrainer_probabilisticOversampling but as my batch size is 2 I'm wondering whether it can help. Is there a way to get the trainer to increase oversampling of the under-represented class?

Thanks,
Daryl.

**Comment by pwaver** [comment]
I have the same question. In other, non self-configuring models I have met this need by means of weighted cross-entropy in the loss function. In its self-configuration, does nnUNet count the class pixels in the labels? If not, is there otherwise a way to give cross-entropy weights? Thank you, Bill

**Reply by pwaver** [reply]
Stepping through the source, I wonder if this is the role of the parameter get_deep_supervision_scales in class nnUNetTrainer, but if it is, I do not see yet how to leverage it. Bill
